### PR TITLE
[Rest Server] Refine hdfs get in container

### PR DESCRIPTION
**Pull request by abuccts**
Refine hdfs get in container. DO NOT download code dir if it is larger than 10MB.

### CODE REVIEW

File: job-tutorial/README.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ Below please find the detailed explanation for each of the parameters in the con
 | `authFile`                     | String, optional, HDFS URI | Docker registry authentication file existing on HDFS |
 | `dataDir`                      | String, optional, HDFS URI | Data directory existing on HDFS          |
 | `outputDir`                    | String, optional, HDFS URI | Output directory on HDFS, `$PAI_DEFAULT_FS_URI/Output/$jobName` will be used if not specified |
-| `codeDir`                      | String, optional, HDFS URI | Code directory existing on HDFS          |
+| `codeDir`                      | String, optional, HDFS URI | Code directory existing on HDFS, should not contain any data and should be less than 10MB     |
```
Comment (sterowang) [inline job-tutorial/README.md]: >10MB  [](start = 149, length = 5)

Is 10M is too small? I suggest 1G would be a more user friendly size. In future, this number should be adjustable from configuration, suggest to add a TODO here to remind this thing.

File: rest-server/src/templates/yarnContainerScript.mustache [hunk 2]
```diff
@@ -184,12 +184,21 @@ export $(grep -v '^#' $ENV_LIST | xargs)
...
+  || { echo "Can not get script from HDFS. HDFS may be down."; exit 1; }
 
 # Prepare user code
 mkdir -p "/tmp/pai-root/code/$APP_ID/$CONTAINER_ID"
 if [[ -n $PAI_CODE_DIR ]]; then
-  hdfs dfs -get $PAI_CODE_DIR "/tmp/pai-root/code/$APP_ID/$CONTAINER_ID" || exit 1
+  hdfs dfs -stat $PAI_CODE_DIR \
+    || { echo "Can not stat $PAI_CODE_DIR"; exit 1; }
+  code_dir_size="$(hdfs dfs -du -s $PAI_CODE_DIR | cut -d ' ' -f 1)"
+  if [ $code_dir_size -gt 10000000 ]; then
```
Comment (fanyangCS) [inline rest-server/src/templates/yarnContainerScript.mustache]: >10000000 [](start = 26, length = 8)

maybe 100M is more appropraite?

**Comment by fanyangCS** [comment]
>   kill 0

can you remove /tmp/pai-root/code/$APP_ID/$CONTAINER_ID/$PAI_CODE_DIR here to make sure?

---
Refers to: rest-server/src/templates/yarnContainerScript.mustache:35 in 4e9aa37. [](commit_id = 4e9aa37b259543601054d2362a70cce7d3494cd4, deletion_comment = False)

File: job-tutorial/README.md (hunk 1 above)
Comment (mzmssg) [reply job-tutorial/README.md]: 10M may be not enough? Jobs may include some demo images or other binary files for preprocess

Comment (fanyangCS) [reply job-tutorial/README.md]: I suggest at least set it to 100M~200M. Remember we not only support DL workloads. Maybe some program has large footprint (like resource files)

---
In reply to: [205726533](https://github.com/Microsoft/pai/pull/975#discussion_r205726533) [](ancestors = 205726533)

**Pull request merged**